### PR TITLE
(doc) Added information about Cake Configuration

### PR DESCRIPTION
**Pull request by gep13**

### CODE REVIEW

File: src/Cake.Web/App_Data/docs/fundamentals/configuration.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+Cake supports the concept of external configuration, to allow the internals of how Cake operates to be controlled from either:
+
+* Environment variables
+* A Configuration File
+* Arguments passed into Cake.exe
+
+For example, when Cake is initially ran, it needs to download the Roslyn assemblies.  It does this by downloading the nuget packages from the default download URL of `https://packages.nuget.org/api/v2`.  However, it may be necessary (for example, when running in an offline/local environment) to download these nuget packaages from an alternative source.  This is where the Cake Configuration comes into play.
+
+By creating an Environment variable with the name of `CAKE_NUGETSOURCE` and setting the value to the URL that is required, Cake will use this alternative download source, rather than the default.
```
Comment (patriksvensson) [inline src/Cake.Web/App_Data/docs/fundamentals/configuration.md]: This should be `CAKE_ROSLYN_NUGETSOURCE`.

File: src/Cake.Web/App_Data/docs/fundamentals/configuration.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Alternatively, you can create a `cake.config` file with the following content:
+
+```sh
+NuGetSource=https://mycustomurl
```
Comment (patriksvensson) [inline src/Cake.Web/App_Data/docs/fundamentals/configuration.md]: This should actually look like:

```
; The configuration file for Cake.

[Roslyn]
NuGetSource=https://mycustomurl
```

File: src/Cake.Web/App_Data/docs/fundamentals/configuration.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+```
+
+Specifying a configuration value within a configuration file will override the same configuration value stored within an equivalent Environment variable.
+
+**NOTE:** This configuration file should be located at the same place as your `build.cake` file.
+
+Finally, you can specify an input parameter directly to the Cake.exe, in the following format:
+
+```sh
+cake.exe --nugetsource=http://mycustomurl
```
Comment (patriksvensson) [inline src/Cake.Web/App_Data/docs/fundamentals/configuration.md]: Should be `cake.exe --roslyn_nugetsource=http://mycustomurl`

File: src/Cake.Web/App_Data/docs/fundamentals/configuration.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+content-type: markdown
+---
+
+Cake supports the concept of external configuration, to allow the internals of how Cake operates to be controlled from either:
```
Comment (patriksvensson) [inline src/Cake.Web/App_Data/docs/fundamentals/configuration.md]: Might be good to point out that the different methods have different priorities.

**Comment by patriksvensson** [comment]
Should we also add some kind of table with built in configurations?
We would have to list the following:

* Environment variable name
* Ini file structure (section and key)
* Argument name

**Comment by gep13** [comment]
@patriksvensson yes, I thought the same about the table of values, but I had thought to leave that until there was more than one option, but thinking about it, I think you are right.

Not sure how the table will render on the site though.  Can I run the website locally to test how it will look?

**Comment by patriksvensson** [comment]
@gep13 Yes, you can. You should also be able to mix HTML with markdown.

**Comment by gep13** [comment]
Thoughts?

image

**Comment by gep13** [comment]
@patriksvensson the idea of a table wasn't really working out, so I have opted for something like this:

image

What do you think?

File: src/Cake.Web/App_Data/docs/fundamentals/default-configuration-values.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+#### Default Value
+
+```sh
+https://packages.nuget.org/api/v2
+```
+
+#### Environment Variable Name
+
+```sh
+CAKE_ROSLYN_NUGET
```
Comment (patriksvensson) [inline src/Cake.Web/App_Data/docs/fundamentals/default-configuration-values.md]: Should be `CAKE_ROSLYN_NUGETSOURCE` right?

**Comment by patriksvensson** [comment]
@gep13 Looks really good. Nice writing! :+1:

File: src/Cake.Web/App_Data/docs/fundamentals/default-configuration-values.md (hunk 5 above)
Comment (gep13) [inline src/Cake.Web/App_Data/docs/fundamentals/default-configuration-values.md]: Doh! Good catch! :+1:

**Comment by gep13** [comment]
@patriksvensson rebased and pushed

**Comment by patriksvensson** [comment]
@gep13 Will merge once CI pass.

**Pull request merged**